### ISSUE TITLE
[SDK] Whether plan to develop an SDK

### ISSUE THREAD

**Issue by vczyh** [vitessio/vitess#8758]
The vtctl return result is not easy to parse. So I call some functions in `soure/go/vt`. 
What is the solution to make vtctl result easy to parse(like `json`). 
I don't find the flag do it. So whether plan to develop an SDK.

**Comment by tokikanno** [comment]
@vczyh: You could use proto files under https://github.com/vitessio/vitess/tree/main/proto

Then you could using any programming language support gRPC for calling vtctl APIs

**Comment by vczyh** [comment]
> @vczyh: You could check proto files under https://github.com/vitessio/vitess/tree/main/proto
> 
> Then you could use any programming language support gRPC for calling vtctl APIs

THX, another question:
- ` go get vitess.io/vitess@aa798b854a0d12db8151157919108eaefb2378b2`
- using some api, for example
```go
server, err := topo.OpenServer("etcd2", strings.Join(addrs, ","), root)
if err != nil {
  return nil, err
}
return server.GetKeyspaces(ctx)
```
- running well under unix, but running into a problem under windows.
![image](https://user-images.githubusercontent.com/33161749/132149832-1dd83e92-69f2-41d0-95df-cc5a543f8aed.png)

go source `log/syslog/syslog.go`
```
// +build !windows,!plan
```

**I wonder how to fix it on windows.**

**Comment by tokikanno** [comment]
1. Did u call the API through gRPC or call through Golang directly?
2. There's no syslog facilities on windows OS, only on un*x like OS. So maybe that's the problem.

**Comment by vczyh** [comment]
> 1. Did u call the API through gRPC or call through Golang directly?
> 2. There's no syslog facilities on windows OS, only on un*x like OS. So maybe that's the problem.

Yes, you're right. I call the API through Golang directly. How to solve it if i continue.

**Comment by tokikanno** [comment]
You should use gRPC as a client program instead of calling internal API by Golang directly.
Since you're just a client, you don't need to deal with the syslog problem (server will).

If you can't sure you're writing a gRPC client. Here is a easy way to identify.

`You will only need files under {Vitess repo}/proto folder for writing a Vitess gRPC  client`

**Comment by ajm188** [comment]
This is covered (incidentally) by https://github.com/vitessio/vitess/projects/9, since that adds a new gRPC interface that _doesn't_ return a json blob but actual structured data for each RPC.
